crates/swc_ecma_minifier/src/analyzer/mod.rs

```diff
@@ -1040,7 +1061,7 @@ where
                 v.mark_indexed_with_dynamic_key();
             }
 
-            if self.ctx.in_assign_lhs {
+            if self.ctx.in_assign_lhs || self.ctx.is_delete_arg {
```

Comment:
I think this is only catching `delete foo.bar`, but fails to capture `delete foo.bar.baz`? Based on the `Expr::Ident(obj) = e.obj` we're inside of. Come to think of it, there are a few others like `foo().bar` that won't be caught?

Reply:
For `delete foo.bar.baz`, `foo` will be marked as property mutated because we don't unset `is_delete_arg` for the `obj` of `MemberExpr`. ( https://github.com/swc-project/swc/blob/114bc97cf5ba430da4b9561613f4657c9b4afeb8/crates/swc_ecma_minifier/src/analyzer/mod.rs#L1032-L1040 )
And currently, we do not inline properties. But nice catch, we should ensure it when we add partial, per-property inlining.